packages/tools/src/tools/ReferenceCursors.ts

```diff
@@ -0,0 +1,424 @@
+    defaultToolProps: ToolProps = {
+      supportedInteractionTypes: ['Mouse', 'Touch'],
+      configuration: {
+        shadow: true,
+        preventHandleOutsideImage: false,
+        displayThreshold: 5,
+        positionSync: true,
+      },
+    }
+  ) {
+    super(toolProps, defaultToolProps);
+  }
+
+  /**
+   * Overwritten mouseMoveCallback since we want to keep track of the current mouse position and redraw on mouseMove
+   * @virtual Event handler for Cornerstone MOUSE_MOVE event.
+   *
+   *
+   * @param evt - The normalized mouse event
+   * @param filteredAnnotations - The annotations to check for hover interactions
+   * @returns True if the annotation needs to be re-drawn by the annotationRenderingEngine.
+   */
+  mouseMoveCallback = (evt: EventTypes.MouseMoveEventType): boolean => {
+    const { detail } = evt;
+    const { element, currentPoints } = detail;
+
+    //save current positions and current element the curser is hovering over
+    this._currentCursorWorldPosition = currentPoints.world;
+    this._currentCanvasPosition = currentPoints.canvas;
+    this._elementWithCursor = element;
+
+    const annotation = this.getActiveAnnotation(element);
+    if (annotation === null) {
+      this.createInitialAnnotation(currentPoints.world, element);
+      return false;
+    }
+    this.updateAnnotationPosition(element, annotation);
+    return false;
+  };
+
+  createInitialAnnotation = (
+    worldPos: Types.Point3,
+    element: HTMLDivElement
+  ): void => {
+    const enabledElement = getEnabledElement(element);
+    if (!enabledElement) throw new Error('No enabled element found');
+    const { viewport, renderingEngine } = enabledElement;
+
+    this.isDrawing = true;
+
+    const camera = viewport.getCamera();
+    const { viewPlaneNormal, viewUp } = camera;
+    if (!viewPlaneNormal || !viewUp) throw new Error('Camera not found');
+
+    const referencedImageId = this.getReferencedImageId(
+      viewport,
+      worldPos,
+      viewPlaneNormal,
+      viewUp
+    );
+
+    const annotation = {
+      highlighted: true,
+      invalidated: true,
+      metadata: {
+        toolName: this.getToolName(),
+        viewPlaneNormal: <Types.Point3>[...viewPlaneNormal],
+        viewUp: <Types.Point3>[...viewUp],
+        FrameOfReferenceUID: viewport.getFrameOfReferenceUID(),
+        referencedImageId,
+      },
+      data: {
+        label: '',
+        handles: {
+          points: [[...worldPos]] as [Types.Point3],
+          activeHandleIndex: null,
+          textBox: {
+            hasMoved: false,
+            worldPosition: <Types.Point3>[0, 0, 0],
+            worldBoundingBox: {
+              topLeft: <Types.Point3>[0, 0, 0],
+              topRight: <Types.Point3>[0, 0, 0],
+              bottomLeft: <Types.Point3>[0, 0, 0],
+              bottomRight: <Types.Point3>[0, 0, 0],
+            },
+          },
+        },
+      },
+    };
+
+    const annotationId = this._addAnnotation(element, annotation);
```

Comment:
This check which is basically checking annotation manager if there is one exists can be done very first thing at the top of this method and avoid all the object and logics up here